plenum/common/util.py

```diff
@@ -580,11 +584,13 @@ def __call__(cls, *args, **kwargs):
         return cls._instances[cls]
 
 
-def isValidEndpoint(endpoint):
-    import ipaddress
+def check_endpoint_valid(endpoint, required: bool=True):
+    if not endpoint and required:
+        raise InvalidEndpointMissing(endpoint)
     ip, port = endpoint.split(':')
     try:
         ipaddress.ip_address(ip)
-        return port.isdigit()
-    except Exception:
-        return False
\ No newline at end of file
+    except Exception as exc:
+        raise InvalidEndpointIpAddress(endpoint) from exc
+    if not port.isdigit():
+        raise InvalidEndpointPort(endpoint)
```

Comment:
where is the return? The function only throws or don't return (returns None implicitly)

This is a major change of behavior, is this correct.